syntax = "proto3";

package tc_serialize;

import "map_renderer.proto";
import "transport_router.proto";

message TransportCatalogue {
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	RenderSettings render_settings = 3;
	TransportRouter transport_router = 4;
}

message Coordinates {
	double lat = 1;
	double lng = 2;
}

message StopsDistance {
	uint32 id = 1;
	int32 distance_between = 2;
}

message Stop {
	bytes name = 1;
	Coordinates coordinates = 2;
	repeated StopsDistance distance = 3;
}

message Bus {
	bytes name = 1;
	repeated uint32 stops_id = 2;
	bool is_circular = 3;
}
